nbextensions/upload_it/main.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+define([
+  'jquery',
+  'base/js/namespace',
+  'base/js/dialog',
+], function($, Jupyter, dialog) {
+  "use strict";
+
+  // Default values for the configuration parameters.
+  let configuration = {
+    "upload_it_server_url": "http://localhost:8000/upload"
+  };
+
+  function initialize() {
+    updateConfig();
+    Jupyter.toolbar.add_buttons_group([
+      Jupyter.keyboard_manager.actions.register({
+        "help": "Upload notebook",
+        "icon": "fa-check-square",
+        "handler": show_upload_dialog
+      }, "upload-notebook", "upload_it")
+    ]);
+  }
+
+  function updateConfig() {
+    // If the notebook has configuration overrides, pick them up.
+    const config = Jupyter.notebook.config;
+    for (let key in configuration) {
+      if (config.data.hasOwnProperty(key)) {
+        configuration[key] = config.data[key];
+      }
+    }
+  }
+
+  // Returns a JQuery object with the dialog.
+  function build_upload_dialog() {
```

Comment:
ditto
https://google.github.io/styleguide/jsguide.html#naming-rules-common-to-all-identifiers